packages/js/product-editor/src/components/custom-fields/edit-modal/edit-modal.tsx

```diff
@@ -101,13 +106,17 @@ export function EditModal( {
 				props.className
 			) }
 		>
-			<TextControl
+			<CustomFieldNameControl
 				ref={ keyInputRef }
 				label={ __( 'Name', 'woocommerce' ) }
-				error={ validationError?.key }
+				allowReset={ false }
+				help={ validationError?.key }
 				value={ customField.key }
 				onChange={ changeHandler( 'key' ) }
 				onBlur={ blurHandler( 'key' ) }
+				className={ classNames( 'woocommerce-product-text-control', {
```

Comment:
[Same](https://github.com/woocommerce/woocommerce/pull/48949/files#r1664912619) with this class name.

Reply:
[e15a530](https://github.com/woocommerce/woocommerce/pull/48949/commits/e15a530cabd2d526104bb8ca286f75b545792cff)